python/tank/platform/validation.py

```diff
@@ -641,57 +641,64 @@ def __validate_settings_hook(self, settings_key, schema, hook_name):
 
         hooks_folder = self._tank_api.pipeline_configuration.get_hooks_location()
 
-        # if setting is default, assume everything is fine
-        if hook_name == constants.TANK_BUNDLE_DEFAULT_HOOK_SETTING:
-            # assume that each app contains its correct hooks
-            return
+        hook_paths_to_validate = []
 
-        elif hook_name.startswith("{self}"):
-            # assume that each app contains its correct hooks
-            return
-        
-        elif hook_name.startswith("{config}"):
-            # config hook 
-            path = hook_name.replace("{config}", hooks_folder)
-            hook_path = path.replace("/", os.path.sep)
-
-        elif hook_name.startswith("{engine}"):
-            # engine hook. see if there is a current engine we can use to
-            # validate against. there should be an engine, but in the case
-            # where validation is being run outside of or before engine
-            # startup, continue and assume the hook exists similar to app
-            # hooks.
-            from .engine import current_engine
-            if current_engine():
-                path = os.path.join(current_engine().disk_location, "hooks")
-                hook_path = path.replace("/", os.path.sep)
-            else:
+        for hook_path in hook_value.split(":"):
```

Comment:
The major change here is to split the value of the hook and validate each path separately. The logic remains the same other than collecting the paths to validate at the end of the method.